### DISCUSSION TITLE
lesson 9 problem in Helperconfig

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#625] [General]
can anybody please help me understanding this function
```
 function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory) {
        if (activeNetworkConfig.vrfCoordinator != address(0)) {
            return activeNetworkConfig;
        }

        uint96 baseFee = 0.25 ether;
        uint96 gasPriceLink = 1e9;
        vm.startBroadcast();
        VRFCoordinatorV2Mock vrfCoordinatorV2Mock = new VRFCoordinatorV2Mock(baseFee,gasPriceLink);

        vm.stopBroadcast();

```

**Comment by shikhar229169** [comment]
Remember we deployed price feeds contract on our local network in order to get the price info.
And for the same reason as we need a coordinator contract on which we will request for random words. For sepolia network chainlink has its own coordinator contract, but for our local network we need to deploy a mock coordinator on which we will place call for requesting random words.

**Reply by shikhar229169** [reply]
Base Fee is the fixed fee in link (which is 0.25 link) that will be taken for each request, as the decimals value for link token is 10<sup>18</sup>, therefore we need to multiply 0.25 by 10<sup>18</sup> and adding ether means we multiply the quantity by 10<sup>18</sup>.

- When you request for random words, the chainlink coordinator will then trigger the fulFilrandomWords function on our Lottery contract and as the chainlink contract is now performing the txn, the gas fee is paid by them, therefore we need to pay this gas to chainlink in link amount. So for every gas spent we pay them link. And gasPriceLink denotes what amount of link we have to pay corresponnding to gas fee paid by chainlink contract to call the fulfilRandomWords on our contract.

**Reply by khushbusandhu** [reply]
okay got your point.  but still, i am not clear for these lines:
```
 function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory) {
        if (activeNetworkConfig.vrfCoordinator != address(0)) {
            return activeNetworkConfig;
        }
```

what does address(0) mean here.

**Reply by shikhar229169** [reply]
address(0) is actually equal to 0x0000000000000000000000000000000000000000.
Now when you call this function, then the vrfCoordinator contract address will not be 0x0000000000000000000000000000000000000000.
So, in case if you again call this function but the coordinator was already deployed, so to prevent multiple deployment of the coordinator contract we use this check, because next time we call it the coordinator will not be at address 0.

**Reply by khushbusandhu** [reply]
okay got it 
thanks again @shikhar229169 :)

**Reply by 0xalexhex** [reply]
as Patrick said the best way to understand code will be to ask ai
